### ISSUE TITLE
kubelet/containerd logs "failed to setup network for sandbox" when using istio-cni 1.9.3 on CentOS7

### ISSUE THREAD

**Issue by funkypenguin** [istio/istio#33138]
(**NOTE**: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>)

**Bug description**

This is a continuation of a [slack thread](https://istio.slack.com/archives/C37A4KAAD/p1622087786382900). In my cluster, as the nodes get busier, istio-proxy-injected pods take longer to start (and sometimes can't start at all), with kubelet/container messages about `failed to setup network for sandbox`. The pods do eventually start, given enough time.

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

When using istio-cni, pods should start up normally

**Steps to reproduce the bug**

* Install istio with `cni.enabled` (my cluster is kubeadm-built, on CentOS7 using kernel 5.11)
* Create a namespace with 'istio-injection=enabled' label applied (I called mine "sandbox")
* Create a simple deployment in this namespace (`kubectl create deployment -n sandbox --image=nginx test`)
* Scale the deployment up enough to make the nodes work a bit
* Observe the kubelet logs, and describe the created pods. You'll note events like this:

```
Events:
  Type     Reason                  Age    From               Message
  ----     ------                  ----   ----               -------
  Normal   Scheduled               3m11s  default-scheduler  Successfully assigned sandbox/test-5f6778868d-28hnw to cow1
  Warning  FailedCreatePodSandBox  3m10s  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "fb4d0911dadea4690e006ab393bb42ebc1547cca154e74dddc9ccefdec81a38d": exit status 2
  Warning  FailedCreatePodSandBox  3m8s   kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "7bab640fc109d1d4d9727c44d96df1847d56bad738e6f39616a9a0a859cf145b": exit status 2
  Normal   Pulling                 3m7s   kubelet            Pulling image "gergorg/proxyv2:1.9.3"
  Normal   Pulled                  3m7s   kubelet            Successfully pulled image "gergorg/proxyv2:1.9.3" in 752.672805ms
  Normal   Created                 3m7s   kubelet            Created container istio-validation
  Normal   Started                 3m6s   kubelet            Started container istio-validation
  Normal   Pulling                 3m6s   kubelet            Pulling image "gergorg/proxyv2:1.9.3"
  Normal   Pulled                  3m5s   kubelet            Successfully pulled image "gergorg/proxyv2:1.9.3" in 788.467888ms
```

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

```
root@cowboy:~# istioctl version --remote
client version: 1.9.3
control plane version: 1.9.3
data plane version: 1.9.3 (156 proxies), 1.9.0 (26 proxies)
root@cowboy:~# kubectl version --short
Client Version: v1.20.5
Server Version: v1.20.5
root@cowboy:~# helm version --short
v3.5.4+g1b5edb6
root@cowboy:~#
```

**How was Istio installed?**

Using istio-operator, which was installed by helm

**Environment where the bug was observed (cloud vendor, OS, etc)**

Bare-metal cluster created with kubeadm, using Calico 3.19 as a CNI

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details (I opted not to do this yet, since I'm not sure whether the data created in the bug report may contain confidential information)

Debug logs from kubelet on one of the nodes are here: 
[debug-logs-cow1.log](https://github.com/istio/istio/files/6551187/debug-logs-cow1.log)

**Comment by funkypenguin** [comment]
Here's the conflist file requested :)

[conflist.txt](https://github.com/istio/istio/files/6551230/conflist.txt)

**Comment by kamboyer** [comment]
Was this confirmed to be a result of Istio CNI? 

Meaning was it confirmed that cluster set ups without Istio or perhaps just without Istio CNI being used do not reproduce this behavior?

If there is more information which directly correlates these observations to Istio CNI, please share.

FYI, we are also observing same reported symptoms on GKE 1.18 & 1.19 (their container optimized OS), containerD & Docker runtimes and those were using Calico 3.16. We had observed failure to create sandbox errors in previous cluster builds (Istio < 1.9.x), but our latest upgrades seem to produce the errors significantly more. These upgrades included moving to GKE 1.19 and to Istio 1.9.2 among other functional changes.

**Comment by bianpengyuan** [comment]
This is likely due to https://github.com/istio/istio/issues/31936. For istio <= 1.10, istio-cni will only run iptables once, and if it gets xtables lock failure, it exists with code 2. At istio 1.11, this should be improved a bit since we added retry. However, it still won't help if the node is super busy and xtables lock is always hold..
